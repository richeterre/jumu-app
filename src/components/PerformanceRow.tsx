import React from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";

import textStyles from "../constants/textStyles";
import { ListPerformanceFragment } from "../graphql/types/generated";
import { flags } from "../helpers/countries";
import { isoTimeToString } from "../helpers/dates";

interface Props {
  performance: ListPerformanceFragment;
  onPress: () => void;
}

const PerformanceRow: React.FC<Props> = props => {
  const {
    performance: {
      stageTime,
      categoryName,
      ageGroup,
      appearances,
      predecessorHost,
    },
    onPress,
  } = props;

  return (
    <TouchableOpacity style={styles.root} onPress={onPress}>
      <Text style={styles.stageTime}>{isoTimeToString(stageTime)}</Text>

      <View style={styles.container}>
        <Text style={styles.categoryInfo}>
          {`${categoryName}, AG ${ageGroup}`}
        </Text>
        <Text style={styles.appearances}>
          {appearances
            .map(a => `${a.participantName}, ${a.instrumentName}`)
            .join("\n")}
        </Text>
        {predecessorHost && (
          <Text style={styles.predecessorHostInfo}>
            {flags(predecessorHost.countryCodes)} {predecessorHost.name}
          </Text>
        )}
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  root: {
    flexDirection: "row",
    paddingHorizontal: 16,
    paddingVertical: 12,
  },
  stageTime: {
    ...textStyles.medium,
    fontWeight: "bold",
  },
  container: {
    flex: 1,
    marginLeft: 16,
  },
  categoryInfo: {
    ...textStyles.medium,
    fontWeight: "bold",
  },
  appearances: {
    ...textStyles.medium,
    marginTop: 8,
  },
  predecessorHostInfo: {
    ...textStyles.medium,
    marginTop: 3,
  },
});

export default PerformanceRow;
