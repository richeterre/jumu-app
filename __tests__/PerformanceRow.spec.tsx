import React from "react";
import renderer from "react-test-renderer";

import PerformanceRow from "../src/components/PerformanceRow";

describe("PerformanceRow", () => {
  test("renders correctly", () => {
    const performanceRow = renderer.create(
      <PerformanceRow
        stageTime="09:45:00"
        categoryInfo="Violine solo, AG II"
        appearances={[
          { participantName: "Theo Trompeter", instrumentName: "Trompete" },
          { participantName: "Paula Pianistin", instrumentName: "Klavier" }
        ]}
      />
    );
    expect(performanceRow).toMatchSnapshot();
  });
});
