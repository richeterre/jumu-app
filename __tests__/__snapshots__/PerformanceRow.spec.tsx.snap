// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerformanceRow renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Lato",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    09:45
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Lato",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Violine solo, AG II
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Lato",
          "fontSize": 16,
          "marginTop": 8,
        }
      }
    >
      Theo Trompeter, Trompete
Paula Pianistin, Klavier
    </Text>
  </View>
</View>
`;
